### PR TITLE
Add view abstraction and configuration import

### PR DESCRIPTION
**Pull request by boucaud**

### CODE REVIEW

**Comment by manthey** [comment]
When I click Import Configuration, I get a warning in the console and it doesn't continue.  The warning and stack trace is:

```
Warning: a promise was created in a handler but was not returned from it, see

  | printWarning | @ | bluebird.js?f684:1590
  | formatAndLogError | @ | bluebird.js?f684:1306
  | warn | @ | bluebird.js?f684:1201
  | Promise._warn | @ | bluebird.js?f684:751
  | checkForgottenReturns | @ | bluebird.js?f684:1176
  | Promise._settlePromiseFromHandler | @ | bluebird.js?f684:3377
  | Promise._settlePromise | @ | bluebird.js?f684:3423
  | Promise._settlePromise0 | @ | bluebird.js?f684:3468
  | Promise._settlePromises | @ | bluebird.js?f684:3548
  | _drainQueueStep | @ | bluebird.js?f684:145
  | _drainQueue | @ | bluebird.js?f684:138
  | Async._drainQueues | @ | bluebird.js?f684:154
  | Async.drainQueues | @ | bluebird.js?f684:67
  | Promise.then (async) |   |  
  | schedule | @ | bluebird.js?f684:4593
  | Async._queueTick | @ | bluebird.js?f684:163
  | AsyncSettlePromises | @ | bluebird.js?f684:128
  | Promise._fulfill | @ | bluebird.js?f684:3494
  | Promise._resolveCallback | @ | bluebird.js?f684:3285
  | resolve | @ | bluebird.js?f684:5016
  | Promise.then (async) |   |  
  | tryCatcher | @ | bluebird.js?f684:5370
  | doThenable | @ | bluebird.js?f684:5006
  | tryConvertToPromise | @ | bluebird.js?f684:4971
  | Promise._resolveCallback | @ | bluebird.js?f684:3284
  | eval | @ | bluebird.js?f684:3338
  | eval | @ | tslib.es6.js?9ab4:69
  | Promise._execute | @ | bluebird.js?f684:1011
  | Promise._resolveFromExecutor | @ | bluebird.js?f684:3337
  | Promise | @ | bluebird.js?f684:2922
  | adopt | @ | tslib.es6.js?9ab4:69
  | step | @ | tslib.es6.js?9ab4:73
  | fulfilled | @ | tslib.es6.js?9ab4:71
  | tryCatcher | @ | bluebird.js?f684:5370
  | Promise._settlePromiseFromHandler | @ | bluebird.js?f684:3366
  | Promise._settlePromise | @ | bluebird.js?f684:3423
  | Promise._settlePromise0 | @ | bluebird.js?f684:3468
  | Promise._settlePromises | @ | bluebird.js?f684:3548
  | _drainQueueStep | @ | bluebird.js?f684:145
  | _drainQueue | @ | bluebird.js?f684:138
  | Async._drainQueues | @ | bluebird.js?f684:154
  | Async.drainQueues | @ | bluebird.js?f684:67
  | Promise.then (async) |   |  
  | schedule | @ | bluebird.js?f684:4593
  | Async._queueTick | @ | bluebird.js?f684:163
  | AsyncSettlePromises | @ | bluebird.js?f684:128
  | Promise._fulfill | @ | bluebird.js?f684:3494
  | Promise._resolveCallback | @ | bluebird.js?f684:3285
  | resolve | @ | bluebird.js?f684:5016
  | Promise.then (async) |   |  
  | tryCatcher | @ | bluebird.js?f684:5370
  | doThenable | @ | bluebird.js?f684:5006
  | tryConvertToPromise | @ | bluebird.js?f684:4971
  | Promise._resolveCallback | @ | bluebird.js?f684:3284
  | eval | @ | bluebird.js?f684:3338
  | eval | @ | tslib.es6.js?9ab4:69
  | Promise._execute | @ | bluebird.js?f684:1011
  | Promise._resolveFromExecutor | @ | bluebird.js?f684:3337
  | Promise | @ | bluebird.js?f684:2922
  | adopt | @ | tslib.es6.js?9ab4:69
  | step | @ | tslib.es6.js?9ab4:73
  | fulfilled | @ | tslib.es6.js?9ab4:71
  | tryCatcher | @ | bluebird.js?f684:5370
  | Promise._settlePromiseFromHandler | @ | bluebird.js?f684:3366
  | Promise._settlePromise | @ | bluebird.js?f684:3423
  | Promise._settlePromise0 | @ | bluebird.js?f684:3468
  | Promise._settlePromises | @ | bluebird.js?f684:3548
  | _drainQueueStep | @ | bluebird.js?f684:145
  | _drainQueue | @ | bluebird.js?f684:138
  | Async._drainQueues | @ | bluebird.js?f684:154
  | Async.drainQueues | @ | bluebird.js?f684:67
  | Promise.then (async) |   |  
  | schedule | @ | bluebird.js?f684:4593
  | Async._queueTick | @ | bluebird.js?f684:163
  | AsyncSettlePromises | @ | bluebird.js?f684:128
  | Promise._fulfill | @ | bluebird.js?f684:3494
  | Promise._resolveCallback | @ | bluebird.js?f684:3285
  | resolve | @ | bluebird.js?f684:5016
  | Promise.then (async) |   |  
  | timerFunc | @ | vue.runtime.esm.js?2b0e:1933
  | nextTick | @ | vue.runtime.esm.js?2b0e:1990
  | queueWatcher | @ | vue.runtime.esm.js?2b0e:4402
  | update | @ | vue.runtime.esm.js?2b0e:4544
  | notify | @ | vue.runtime.esm.js?2b0e:730
  | reactiveSetter | @ | vue.runtime.esm.js?2b0e:1055
  | eval | @ | vue-router.esm.js?8c4f:2919
  | eval | @ | vue-router.esm.js?8c4f:2918
  | updateRoute | @ | vue-router.esm.js?8c4f:2308
  | eval | @ | vue-router.esm.js?8c4f:2156
  | eval | @ | vue-router.esm.js?8c4f:2294
  | step | @ | vue-router.esm.js?8c4f:1895
  | step | @ | vue-router.esm.js?8c4f:1902
  | runQueue | @ | vue-router.esm.js?8c4f:1906
  | eval | @ | vue-router.esm.js?8c4f:2289
  | step | @ | vue-router.esm.js?8c4f:1895
  | eval | @ | vue-router.esm.js?8c4f:1899
  | eval | @ | vue-router.esm.js?8c4f:2274
  | eval | @ | vue-router.esm.js?8c4f:2055
  | iterator | @ | vue-router.esm.js?8c4f:2252
  | step | @ | vue-router.esm.js?8c4f:1898
  | step | @ | vue-router.esm.js?8c4f:1902
  | eval | @ | vue-router.esm.js?8c4f:1899
  | eval | @ | vue-router.esm.js?8c4f:2274
  | eval | @ | routeMapper.ts?9f22:61
  | Promise.then (async) |   |  
  | beforeRouteUpdate | @ | routeMapper.ts?9f22:61
  | boundRouteGuard | @ | vue-router.esm.js?8c4f:2399
  | iterator | @ | vue-router.esm.js?8c4f:2252
  | step | @ | vue-router.esm.js?8c4f:1898
  | step | @ | vue-router.esm.js?8c4f:1902
  | runQueue | @ | vue-router.esm.js?8c4f:1906
  | confirmTransition | @ | vue-router.esm.js?8c4f:2282
  | transitionTo | @ | vue-router.esm.js?8c4f:2152
  | push | @ | vue-router.esm.js?8c4f:2622
  | push | @ | vue-router.esm.js?8c4f:2953
  | handler | @ | vue-router.esm.js?8c4f:1102
  | invokeWithErrorHandling | @ | vue.runtime.esm.js?2b0e:1854
  | invoker | @ | vue.runtime.esm.js?2b0e:2179
  | original._wrapper | @ | vue.runtime.esm.js?2b0e:6917
  | Show 40 more frames
```

**Comment by boucaud** [comment]
Thank you, I'm looking into it.

Another question: right now I'm simply looking for all datasets in the current user's public directory. Should this be done for all users instead ?

**Comment by arjunrajlab** [comment]
I think probably it makes sense to look through all users for now, so that you can use configurations that others have generated. At some point we may want to organize by groups of users and limit scope to that (if we end up with hundreds-thousands of users), but I think that is in the distant future for now.

**Comment by boucaud** [comment]
I have updated the branch to fix that warning issue, and look through all users.

File: src/store/GirderAPI.ts [hunk 1]
```diff
@@ -64,10 +66,53 @@ export default class GirderAPI {
...
+      `folder?parentType=folder&parentId=${publicFolder._id}`
+    );
+    if (result.status !== 200) {
+      throw new Error(
+        `Could not get a list of datasets for folder ${publicFolder.name}: ${result.status}: ${result.statusText}`
+      );
+      return [];
+    }
+    if (result.data?.length) {
+      const folders = result.data;
```
Comment (manthey) [inline src/store/GirderAPI.ts:104]: This should probably be filtered here for only folders that are actually datasets:
```
folders = folders.filter((folder: IGirderFolder) => (folder.meta || {}).subtype === 'contrastDataset');
```
(I don't know if that will pass linting or typescript checks)

**Review by manthey — APPROVED** [review]

**Pull request merged**